models/users.js

```diff
@@ -572,6 +572,41 @@ const fetchAllUsers = async () => {
   return users;
 };
 
+const archiveUserIfNotInDiscord = async () => {
+  try {
+    const snapshot = await userModel.where("roles.in_discord", "==", false).get();
+    const batch = firestore.batch();
+    const summary = {
+      totalUsersWithArchivedRoleUpdated: 0,
+    };
+
+    snapshot.forEach((user) => {
+      const id = user.id;
+      const userData = user.data();
+      const isArchived = userData?.roles?.archived;
+
+      if (!isArchived) {
+        const updatedUserData = {
+          ...userData,
+          roles: {
+            ...userData.roles,
+            archived: true,
+          },
+        };
+
+        batch.update(userModel.doc(id), updatedUserData);
```

Comment:
batch has a limit of 500. can you also handle the case of more than 500 documents.